src/table.js

```diff
@@ -1322,6 +1322,87 @@ Table.prototype.sampleRowKeysStream = function(gaxOptions) {
   ]);
 };
 
+/**
+ * Generates Consistency-Token and check consistency for generated token
+ * In-case consistency check returns false, retrial is done in interval
+ * of 5 seconds till 10 minutes, after that it returns false.
+ *
+ * @param {function(?error, ?boolean)} callback The callback function.
+ * @param {?Error} callback.err An error returned while making this request.
+ * @param {?Boolean} callback.resp Boolean value.
+ */
+Table.prototype.waitForReplication = function(callback) {
+  const self = this;
+
+  const reqOpts = {
+    name: self.id,
+  };
+
+  self.bigtable.request(
+    {
+      client: 'BigtableTableAdminClient',
+      method: 'generateConsistencyToken',
+      reqOpts: reqOpts,
+    },
+    (err, resp) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+
+      const timeoutAfterTenMinutes = setTimeout(() => {
+        callback(null, false);
+      }, 10 * 60 * 1000);
+
+      function retryIfNecessary(err, res) {
+        if (!err && res === true) {
+          clearTimeout(timeoutAfterTenMinutes);
+          callback(null, true);
+          return;
+        }
+        setTimeout(checkConsistency, 5000);
+      }
+
+      function checkConsistency() {
```

Comment:
using checkConsistency for this inner function as well as the table method is confusing, rename this to something like recheck or poll

Reply:
renamed function to launchCheck